modin/pandas/groupby.py

```diff
@@ -368,16 +484,19 @@ def __getitem__(self, key):
                 " is not yet supported."
             )
         return SeriesGroupBy(
-            self._df[key],
+            self._df,
```

Comment:
pandas sometimes provides a subset of the columns. Can we imitate same behavior?
https://github.com/pandas-dev/pandas/blob/098661e8773dc2026f89ec39ca87625c2fb870dd/pandas/core/base.py#L233

Reply:
We can, but I don't see any benefits of this. If the selection is a 1-dim, we can't get the Series as a source object for groupby, because we also have to select the "by" data and the resulted object won't be a Series anyway:
```python
return SeriesGroupBy(
    self._df[key + by_columns],
    ...
)
```